src/main/resources/liquibase/crop_changelog/21_3_0.xml

```diff
@@ -0,0 +1,814 @@
+
+			<!-- Remove ENTRY_CODE as static column -->
+			DELETE FROM list_data_view WHERE static_id = 8300;
+
+			<!-- Migrate ENTRY_CODE from listdata to list_data_details -->
+			INSERT INTO list_data_details(variable_id, lrecid, value, created_by)
+			SELECT 8300, r.lrecid, r.entrycd, ln.listuid FROM listdata r inner join listnms ln on ln.listid = r.listid;
+
+			<!-- Insert ENTRY_CODE variable to list_data_view -->
+			INSERT INTO list_data_view(listid,cvterm_id,type_id)
+			SELECT distinct(ld.listid), 8300, 1815 FROM listnms ld;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-30">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+				SELECT COUNT(1) > 0
+				FROM project p
+						 INNER JOIN projectprop pp ON p.project_id = pp.project_id AND p.dataset_type_id = 4
+						 INNER JOIN stock s ON p.parent_project_id = s.project_id
+						 INNER JOIN stockprop sp ON s.stock_id = sp.stock_id
+				WHERE pp.variable_id = 8300 AND sp.type_id = 8300;
+            </sqlCheck>
+		</preConditions>
+		<comment>ENTRY_CODE migration as stock property</comment>
+		<sql>
+			<!-- Only migrate the ENTRY_CODE that the user added in the study -->
+			INSERT INTO stockprop(stock_id, type_id, value)
+				SELECT stock_id, 8300, s.value FROM project p
+						 INNER JOIN projectprop pp ON p.project_id = pp.project_id AND p.dataset_type_id = 4
+						 INNER JOIN stock s ON p.parent_project_id = s.project_id
+				WHERE pp.variable_id = 8300 AND s.value IS NOT NULL AND s.value != '';
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-31">
+        <preConditions onFail="MARK_RAN">
+			<columnExists tableName="stock" columnName="value"/>
+		</preConditions>
+		<comment>Rename column 'value' to 'value_bkp' in table 'stock'</comment>
+		<sql>
+			ALTER TABLE stock CHANGE value value_bkp varchar(255) null;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-32">
+        <preConditions onFail="MARK_RAN">
+			<columnExists tableName="stock" columnName="type_id"/>
+		</preConditions>
+		<comment>Rename column 'type_id' to 'type_id_bkp' in table 'stock' and make it nullable</comment>
+		<sql>
+			ALTER TABLE stock CHANGE type_id type_id_bkp int null;
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="v21.3.0-33">
+		<preConditions onFail="MARK_RAN">
+			<columnExists tableName="listdata" columnName="entrycd"/>
+		</preConditions>
+		<comment>Rename column 'entrycd' to 'entrycd_bkp' in table listdata and make it nullable.</comment>
+		<sql dbms="mysql" splitStatements="true">
+		ALTER TABLE listdata
+			CHANGE COLUMN entrycd entrycd_bkp VARCHAR (47) NULL DEFAULT '-';
+		</sql>
+	</changeSet>
+	<!-- End of ENTRY_CODE related changesets -->
+
+	<changeSet author="cuenyad" id="v21.3.0-34">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                SELECT COUNT(*) > 0 FROM CVTERM WHERE IS_SYSTEM = 1
```

Comment:
@cuenyad is_system = 0 should be the right query here. It is currently running because we have some of the variables in the list already marked as system variables.

Reply:
done!.